src/Ticket.php

```diff
@@ -1547,9 +1547,19 @@ public function computeTakeIntoAccountDelayStat()
             isset($this->fields['id'])
             && !empty($this->fields['date'])
         ) {
+	   // If SLA TTO calendar is defined use it for Take Into Account, if not, fallback to previous method (SLA TTR)
             $calendars_id = $this->getCalendar();
             $calendar     = new Calendar();
-
+            if (isset($this->fields['slas_id_tto']) && $this->fields['slas_id_tto'] > 0){
+                    $sla = new SLA();
+                    if($sla->getFromDB($this->fields['slas_id_tto'])){
+                        if($sla->fields['use_ticket_calendar']){
+                                $calendars_id = parent::getCalendar();
+                        } else {
+                                $calendars_id = $sla->getField('calendars_id');
+                        }
+                    }
+            }
```

Comment:
I may be wrong, but I guess it could be moved into `Ticket::getCalendar()` using following rules

- if ticket has not yet been take into account (i.e. `takeintoaccount_delay_stat` is 0, I guess) -> try to find calendar attached to TTO SLA
- fallback to TTR SLA
- fallback to `parent::getCalendar()`

@flegastelois would this be correct ?

Reply:
Maybe just using additional parameter as ``$this->getCalendar()`` is used in ``CommonITILObject::computeSolveDelayStat`` and ``CommonITILObject::computeCloseDelayStat`` and come back from waiting to working status. Each of them should refer to TTR SLA regardless whether Ticket has been taken to account or not, there are are such rare (and maybe incorrect) cases.

Reply:
Indeed, introducting a `string $type = 'ttr'` argument on `Ticket::getCalendar()` method, as all usages except the one affected by current PR should always refer to TTR SLA calendar.

Reply:
Updated Ticket::getCalendar() to requested calendar type.
Also I did not make computeTakeIntoAccountDelayStat to fallback to SLM::TTR in case SLM::TTO calednar is 0, but it uses what is returned so parent::getCalendar()